presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -1524,6 +1529,11 @@ private PlanNode visitNonFilteringSemiJoin(SemiJoinNode node, RewriteContext<Row
             return output;
         }
 
+        private boolean isEnableDynamicFiltering()
+        {
+            return !nativeExecution && SystemSessionProperties.isEnableDynamicFiltering(session);
+        }
```

Comment:
@mbasmanova 
I wonder if that would be better to have that check in the SystemSessionProperties itself?

Reply:
That's a good question. SystemSessionProperties.isEnableDynamicFiltering is a static method, so doesn't have access to FeaturesConfig.

@tdcmeehan Tim, what do you think?

Reply:
Practically, you could inject an instance of `SystemSessionProperties` and create a non-static method which has access to the feature config.  But IMO it's just simpler and more straightforward to do it as it's written here.